Hearthstone/app/src/test/java/com/example/hearthstone/ui/searchpage/SearchPageViewModelTest.kt

```diff
@@ -0,0 +1,310 @@
+package com.example.hearthstone.ui.searchpage
+
+import android.app.Application
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.viewModelScope
+import com.example.hearthstone.data.model.HSCardsByClassModel
+import com.example.hearthstone.data.model.Result
+import com.example.hearthstone.data.network.repo.HSRepoImpl
+import com.example.hearthstone.database.dao.HearthstoneDAO
+import com.example.hearthstone.database.model.HearthstoneEntity
+import com.example.hearthstone.dialogue.ErrorDialog
+import com.example.hearthstone.ui.homepage.HomeViewModelTest
+import io.mockk.*
+import junit.framework.TestCase.assertEquals
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SearchPageViewModelTest {
+    @get:Rule
+    var instantTaskExecutor = InstantTaskExecutorRule()
+
+    private lateinit var testViewModel: SearchPageViewModel
+    private val mockApp = mockk<Application>(relaxed = true)
+    private val mockRepoImpl = mockk<HSRepoImpl>(relaxed = true)
+    private val mockDB = mockk<HearthstoneDAO>(relaxed = true)
+    private val testDispatcher = StandardTestDispatcher()
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+        testViewModel = SearchPageViewModel(mockApp, mockRepoImpl, mockDB)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+        unmockkAll()
+    }
+
+
+    @Test
+    fun `getCardsByClass should return list of cards`() {
+        runTest {
+            //Given
+            coEvery { mockRepoImpl.getCardsByClass(any()) } returns expectedTestList
+
+            //When
+            testViewModel.getCardsByClass("Mage")
+            (testViewModel.viewModelScope.coroutineContext[Job]?.children?.forEach { it.join() })
+        }
+        //Then
+        assertEquals(4, testViewModel.cards.value?.size)
```

Comment:
you should be checking at least one item in your list. Maybe even check the type of object in that list as well